adapter.go

```diff
@@ -109,10 +120,13 @@ func createCasbinDatabase(arg interface{}) (*pg.DB, error) {
 	db := pg.Connect(opts)
 	defer db.Close()
 
-	_, err = db.Exec("CREATE DATABASE casbin")
+	_, err = db.Exec(fmt.Sprintf("CREATE DATABASE %s", dbname))
+	if err != nil && !strings.Contains(fmt.Sprintf("%s", err), "42P04") {
```

Comment:
Why not just if err != nil？

Reply:
@tangyang9464 maybe similar to: https://github.com/search?q=org%3Acasbin+42P04&type=code ?

Reply:
@JalinWang

Reply:
> @tangyang9464 maybe similar to: https://github.com/search?q=org%3Acasbin+42P04&type=code ?

Yep. Postgres doesn't support `CREATE DATABASE IF NOT EXISTS`, and I handle the error as gorm-adapter does. @tangyang9464

Reply:
@JalinWang I mean, if there are other errors , Shouldn't it be handled? Why use ``&&``

Reply:
@JalinWang I mean, if there are other errors , Shouldn't it be handled? Why use ``&&``

Reply:
> @JalinWang I mean, if there are other errors , Shouldn't it be handled? Why use `&&`

@tangyang9464 
|                     | err != nil | !strings.Contains(fmt.Sprintf("%s", err), "42P04") | consequence(&&) |
| ------------------- | ---------- | -------------------------------------------------- | --------------- |
| no error            | false      |                                                    | continue        |
| other error         | true       | ! false = true                                     | return nil, err |
| duplicated db error | true       | ! true = false                                     | continue        |

Reply:
Got it. It's ok